python/cudf/cudf/tests/test_indexing.py

```diff
@@ -1683,3 +1683,11 @@ def test_dataframe_indexing_setitem_np_cp_array(array, is_error):
             "(10, 3) could not be broadcast to indexing "
             "result of shape (10, 2)",
         )
+
+
+def test_iloc_single_row_with_nullable_column():
+    # see https://github.com/rapidsai/cudf/issues/11349
+    pdf = pd.DataFrame({"a": [0, 1, 2, 3], "b": [0.1, 0.2, None, 0.4]})
+    df = cudf.from_pandas(pdf)
+
+    assert_eq(pdf.iloc[0], df.iloc[0])
```

Comment:
I'm almost inclined to just make this 
```suggestion
    df.iloc[0]
```
The assertion obscures what we're really testing, which is that we have a valid, indexable object at the end.

Reply:
I would check that the result is correct too

Reply:
I think it doesn't hurt to do both?

Reply:
FWIW, here's a slightly more minimal exhibitor of the issue:
```
from cudf import _lib as libcudf

column = cudf.Series([1.0, None])[:1]._column
transpose_column = libcudf.transpose.transpose(column) => segfault
```

Reply:
I have no problem with checking both, but I want it to be obvious why this is different from every other `iloc` test without people having to open their browser to the GH issue. Otherwise this test just looks redundant (assuming that people don't read the issue, which probably is true about half the time).

Reply:
Fix'd.

Honestly, I think `iloc` just  _happened_ to be where this bug manifested and this isn't the most appropriate place to test this fix. At the same time, I can't think of any test of our public API that would reliably test against failures like this.

One more reason to have a stable, tested, public, set of wrappers around `libcudf` :)

Reply:
Now that I started that conversation it seems like we come up with new reasons every day :) All the more reason to get started around that sooner!